app/src/main/java/com/sun/colornotetaking/ui/adapter/TaskAdapter.java

```diff
@@ -0,0 +1,98 @@
+package com.sun.colornotetaking.ui.adapter;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Filter;
+import android.widget.Filterable;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.sun.colornotetaking.R;
+import com.sun.colornotetaking.data.model.Task;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskAdapter extends BaseAdapter<Task, TaskAdapter.TaskHolder> implements Filterable {
+
+    private List<Task> mSearchList;
+
+    public TaskAdapter(Context context, List<Task> data) {
+        super(context, data);
+        mSearchList = new ArrayList<>(data);
+    }
+
+    @NonNull
+    @Override
+    public TaskHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
+        View view = LayoutInflater.from(getContext()).inflate(R.layout.main_task_item_cardview, viewGroup, false);
+        return new TaskHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull TaskHolder taskHolder, int i) {
+        taskHolder.bindView(getData().get(i));
+    }
+
+    @Override
+    public Filter getFilter() {
+        return mTaskFilter;
+    }
+
+    class TaskHolder extends RecyclerView.ViewHolder {
```

Comment:
```suggestion
    static class ViewHolder extends RecyclerView.ViewHolder {
```

ViewHolder nên để là nested class chứ k nên để là inner class
vì sẽ xảy ra memory leak